tests/rptest/e2e_tests/flink_scale_test.py

```diff
@@ -82,54 +88,140 @@ def assert_failed_jobs(job_list):
         self.flink.start()
 
         # Load python workload to target node
+        # Hardcoded file
         # TODO: Add workload config management
+        # Currently, each INSERT operator will generate 1 subjob
+        # So this config will generate 256 / 64 jobs
         _workload_config = {
             "log_level": "DEBUG",
             "brokers": self.redpanda.brokers(),
             "producer_group": "flink_group",
             "consumer_group": "flink_group",
-            "topic_name": self.topic_name,
+            "topic_name": "flink_scale_topic_placeholder",
             "mode": "produce",
-            "count": 65535 * 32
+            "count": total_events // topics_per_node
         }
 
         # Get workload
         workloads = self.workload_manager.get_workloads(
             ['flink', 'transactions', 'scale'])
         # Run produce part
-        all = self._run_workloads(workloads, _workload_config)
-        # Assert failed jobs
-        assert_failed_jobs(all)
-        # Wait to finish
-        self.flink.wait(timeout_sec=1800)
+        all_jobs = []
+        for spec in self.topic_specs:
+            _workload_config['topic_name'] = spec.name
+            self.logger.debug("Submitting job with config: \n"
+                              f"{json.dumps(_workload_config, indent=2)}")
+            all_jobs += self._run_workloads(workloads, _workload_config)
 
-        # Run consume part
-        _workload_config['mode'] = "consume"
-        all = self._run_workloads(workloads, _workload_config)
         # Assert failed jobs
-        assert_failed_jobs(all)
+        assert_failed_jobs(all_jobs)
 
         # Wait to finish
         self.flink.wait(timeout_sec=1800)
 
-        # TODO: Validate according to metrics
-        # vectorized_internal_rpc_latency_sum{method="commit_tx",service="tx_gateway",shard="1"} 20029
-        # vectorized_internal_rpc_latency_sum{method="commit_tx",service="tx_gateway",shard="2"} 11483
-        # vectorized_internal_rpc_latency_sum{method="commit_tx",service="tx_gateway",shard="3"} 21542
+        topic_watermarks = []
+        for spec in self.topic_specs:
+            hwm = get_hwm_for_topic(spec.name)
+            topic_watermarks.append(hwm)
+
+        # Print out in one place:
+        for idx in range(len(topic_watermarks)):
+            self.logger.info(f"Topic: {self.topic_specs[idx].name}, "
+                             f"watermark: {topic_watermarks[idx]}")
+        self.logger.info(
+            f"Total messages/High watermark sum: {sum(topic_watermarks)}")
+
+        # Simple validation according to RP topic watermarks
+        sum_hwm = sum(topic_watermarks)
+        assert sum_hwm >= total_events, \
+            "High watermark is less than targer total events: " \
+            f"{sum_hwm} hwm, {total_events} target total"
+
+        # Calculate per-node vectorized_internal_rpc_latency_sum/commit_tx
+        def current_rpc_latency_sum(method, nodes) -> list[MetricSamples]:
```

Comment:
nit: I'd rename to total_commit_requests or something

current_rpc_latency_sum is the metric we are relying on but has nothing to do with transactions, I'd rename to make it more readable.

Reply:
Agreed